tests/fuzz/run.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+import argparse
+import logging
+import os
+import re
+import subprocess
+
+from concurrent.futures import ProcessPoolExecutor, as_completed
+
+
+def job(command):
+    """Takes a list of str, and runs it as a subprocess."""
+    command_line = " ".join(command)
+    logging.debug(f"Running '{command_line}'\n")
+    res = subprocess.run(command, check=True, stderr=subprocess.PIPE,
+                         universal_newlines=True).stderr
+    logging.debug(f"Command '{command_line} output:\n'{res}'")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Run the fuzz targets a given amount of times, or generate"
+                    " new seeds.",
+    )
+    parser.add_argument(
+        "seed_dir",
+        help="The parent directory for the seed corpora of each target.",
+    )
+    parser.add_argument("-g", "--generate", action="store_true")
+    parser.add_argument(
+        "-j",
+        "--par",
+        type=int,
+        default=1,
+        help="How many targets to run in parallel.",
+    )
+    parser.add_argument(
+        "-n",
+        "--runs",
+        default=100000,
+        help="How many times to run each target (if generating).",
+    )
+    parser.add_argument(
+        "-m",
+        "--merge_dir",
+        default=None,
+        help="The parent directory to merge each target corpus from.",
+    )
+
+    args = parser.parse_args()
+
+    logging.basicConfig(level=logging.DEBUG)
+
+    target_dir = os.path.abspath(os.path.dirname(__file__))
+    targets = [os.path.join(target_dir, f) for f in os.listdir(target_dir)
+               if re.compile(r"^fuzz-[\w-]*$").findall(f)]
+    with ProcessPoolExecutor(max_workers=args.par) as pool:
```

Comment:
I wonder if using `multiprocessing.Pool(args.par).map` or a `ThreadPoolExecutor.map` wouldn't be faster and less bookkeeping overall.

Reply:
Will try (made sense to me though to have a **process** per target)

Reply:
Well, if we are using a `ProcessPoolExecutor` we are actually using two processes for each fuzzer:

 - The worker process that communicates with the main process, and spawns the subprocess
 - The subprocess which `execv`s the fuzz binary.

We don't really gain anything from spinning up the worker process.

Reply:
Yeah yeah definitely

Reply:
Made it a `ThreadPoolExecutor`